nifi-extension-bundles/nifi-poi-bundle/nifi-poi-services/src/main/java/org/apache/nifi/excel/StandardCellFieldTypeReader.java

```diff
@@ -0,0 +1,152 @@
+            // Date Formatting check limited to NUMERIC Cell Types
+            if (DateUtil.isCellDateFormatted(cell)) {
+                dataType = getDateTimeDataType(cell);
+            } else {
+                final double numericCellValue = cell.getNumericCellValue();
+                final long roundedCellValue = (long) numericCellValue;
+                if (roundedCellValue == numericCellValue) {
+                    dataType = RecordFieldType.LONG.getDataType();
+                } else {
+                    // Default to DOUBLE for NUMERIC values following cell.getNumericCellValue()
+                    dataType = RecordFieldType.DOUBLE.getDataType();
+                }
+            }
+        } else if (CellType.BOOLEAN == cellType) {
+            dataType = RecordFieldType.BOOLEAN.getDataType();
+        } else if (CellType.STRING == cellType) {
+            final String cellValue = cell.getStringCellValue();
+            // Attempt Time Value inference for STRING cell values
+            final Optional<DataType> timeDataType = timeValueInference.getDataType(cellValue);
+            dataType = timeDataType.orElse(RecordFieldType.STRING.getDataType());
+        } else if (CellType.FORMULA == cellType) {
+            dataType = getFormulaResultDataType(cell);
+        } else {
+            // Default to null for known and unknown Cell Types
+            dataType = null;
+        }
+
+        return dataType;
+    }
+
+    private DataType getDateTimeDataType(final Cell cell) {
+        final DataType dataType;
+
+        final double numericCellValue = cell.getNumericCellValue();
+        final long roundedCellValue = (long) numericCellValue;
+        if (roundedCellValue == numericCellValue) {
+            // Numbers without decimal fractions indicate Dates without Times
+            dataType = RecordFieldType.DATE.getDataType();
+        } else if (numericCellValue < 1) {
+            // Decimal fractions indicate Times without Dates
+            dataType = RecordFieldType.TIME.getDataType();
+        } else {
+            dataType = RecordFieldType.TIMESTAMP.getDataType();
+        }
+
+        return dataType;
+    }
+
+    private DataType getFormulaResultDataType(final Cell cell) {
+        final DataType dataType;
+
+        final CellType formulaResultType = cell.getCachedFormulaResultType();
+        if (CellType.BOOLEAN == formulaResultType) {
+            dataType = RecordFieldType.BOOLEAN.getDataType();
+        } else if (CellType.STRING == formulaResultType) {
+            dataType = RecordFieldType.STRING.getDataType();
+        } else if (CellType.NUMERIC == formulaResultType) {
+            // Date Formatting check limited to NUMERIC Cell Types without Conditional Formatting Evaluator
+            if (DateUtil.isCellDateFormatted(cell)) {
+                // Default to TIMESTAMP for date formatted values using standard Date Time Pattern
+                dataType = RecordFieldType.TIMESTAMP.getDataType();
```

Comment:
I do not see any methods which can get the cached formula result but do we leave this as `TIMESTAMP` or somehow try to get the result and determine whether its a `DATE` or `TIME` like we did above?

Reply:
I am not seeing anything directly, its probably best to leave as is.

Reply:
Sounds good, we can revisit in the future based on potential issues if needed.

Reply:
Looking at the DataFormatter code, specifically lines 513-522

 ```
synchronized(dateFormat) {
                if (dateFormat instanceof ExcelStyleDateFormatter) {
                    ((ExcelStyleDateFormatter)dateFormat).setDateToBeFormatted(cell.getNumericCellValue());
                }

                Date d = cell.getDateCellValue();
                return this.performDateFormatting(d, (Format)dateFormat);
            }
```

it seems it gets the cell's numeric or date value. If that is the case then I assume you can reuse the `getDateTimeDataType` above to determine the precise date type.

Reply:
Thanks for the additional pointer. I ran a test with a formula cell in XLSX, confirmed the numeric cell value was available, and I was able to reuse the same `getDateTimeDataType()` method to narrow the Data Type detection, as already used with numeric types. This also aligns with the behavior of `getFormulaCellValue()` in `ExcelRecordReader`.

Reply:
Thanks for pointing out that this aligns to `getFormulaCellValue()`